webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/BinaryDataController.scala

```diff
@@ -191,6 +200,263 @@ class BinaryDataController @Inject()(
+            "order" -> "F",
+            "chunks" -> List(channels, cubeLength, cubeLength, cubeLength),
+            "compressor" -> compressor,
+            "filters" -> None,
+            "shape" -> List(
+              channels,
+              (dataLayer.boundingBox.width + dataLayer.boundingBox.topLeft.x) / parsedMag.x,
+              (dataLayer.boundingBox.height + dataLayer.boundingBox.topLeft.y) / parsedMag.y,
+              (dataLayer.boundingBox.depth + dataLayer.boundingBox.topLeft.z) / parsedMag.z
+            ),
+            "dimension_seperator" -> "/" // This doesn't seem to work effectively...
+          ))
+    }
+  }
+
+  private def getTokenFromHeader(token: Option[String], request: Request[AnyContent]) =
+    token or request.headers.get("X-Auth-Token")
+
+  private def parseMagIfExists(layer: DataLayer, mag: String): Option[Vec3Int] = {
+    val singleRx = "\\s*([0-9]+)\\s*".r
+    val longMag = mag match {
+      case singleRx(x) => "%s-%s-%s".format(x, x, x)
+      case _           => mag
+    }
+    val parsedMag = Vec3Int.fromForm(longMag)
+    Some(parsedMag).filter(layer.containsResolution)
+  }
+
+  private def zarrDtypeFromElementClass(elementClass: ElementClass.Value): (Int, String) = elementClass match {
+    case ElementClass.uint8  => (1, "<u1")
+    case ElementClass.uint16 => (1, "<u2")
+    case ElementClass.uint24 => (3, "<u1")
+    case ElementClass.uint32 => (1, "<u4")
+    case ElementClass.uint64 => (1, "<u8")
+    case ElementClass.float  => (1, "<f4")
+    case ElementClass.double => (1, "<f8")
+    case ElementClass.int8   => (1, "<i1")
+    case ElementClass.int16  => (1, "<i2")
+    case ElementClass.int32  => (1, "<i4")
+    case ElementClass.int64  => (1, "<i8")
+  }
+
+  private def isWkwCompressed(mag: Vec3Int,
+                              dataSource: DataSource,
+                              dataLayer: DataLayer,
+                              dataLayerName: String): Fox[Boolean] = {
+    val result = dataLayer.bucketProvider match {
+      case provider: WKWBucketProvider =>
+        WKWHeader(
+          provider
+            .wkwHeaderFilePath(mag, Some(dataSource.id), Some(dataLayerName), binaryDataService.dataBaseDir)
+            .toFile)
+      case provider: ZarrBucketProvider => ???
+    }
+
+    result.flatMap((h) => Full(h.isCompressed))
+  }
+
+  private def isWkwLZ4Compressed(mag: Vec3Int,
+                                 dataSource: DataSource,
+                                 dataLayer: DataLayer,
+                                 dataLayerName: String): Fox[Boolean] =
+    dataLayer.bucketProvider match {
+      case provider: WKWBucketProvider =>
+        WKWHeader(
+          provider
+            .wkwHeaderFilePath(mag, Some(dataSource.id), Some(dataLayerName), binaryDataService.dataBaseDir)
+            .toFile).flatMap((h) => Full(h.blockType == BlockType.LZ4))
+      case _: ZarrBucketProvider => ???
+    }
+
+  def requestRawZarrWithDot(
```

Comment:
Why do we need to implement both routes (dot and slash)? I’d say we are in control which separator we want (since we generate the .zarray) – so we should be allowed to pick one, right?

Reply:
This was a hacky fix, because the client ignored the specified slash. Since the dot route is working fine, I will remove the slash route